File: frontend/src/types/finding.ts
export type FindingSeverity = 'CRITICAL' | 'HIGH' | 'MEDIUM' | 'LOW';

export type FindingCategory =
  | 'Security'
  | 'Testing'
  | 'Code Quality'
  | 'Dependencies'
  | 'Performance'
  | 'CODE_QUALITY'
  | 'TESTING'
  | 'DEPENDENCY'
  | 'DEPENDENCIES'
  | 'SECURITY'
  | 'PERFORMANCE';

export interface Finding {
  id?: string;
  analysisId?: string;
  releaseId?: string;
  title: string;
  category: FindingCategory;
  severity: FindingSeverity;
  filePath: string;
  lineNumber?: number;
  description: string;
  codeSnippet?: string;
  highlightedLine?: number;
  whatIsWrong?: string;
  whyItMatters?: string;
  whatToReview?: string;
  recommendedAction?: string;
  aiExplanation?: string;
  ruleId?: string;
  evidence?: string;
  confidence?: 'HIGH' | 'MEDIUM' | 'LOW' | string;
  impact?: string;
  status?: string;
}

File: backend/src/main/java/com/aireadiness/model/Finding.java
package com.aireadiness.model;

public class Finding {

    private String id;
    private String analysisId;
    private String category; // CODE_QUALITY, TESTING, DEPENDENCY, SECURITY, PERFORMANCE
    private String ruleId;
    private String severity; // INFO, LOW, MEDIUM, HIGH, CRITICAL
    private String title;
    private String description;
    private String filePath;
    private Integer lineNumber;
    private String evidence;
    private String confidence; // HIGH, MEDIUM, LOW
    private String impact;
    private String status = "OPEN"; // OPEN, ACKNOWLEDGED, RESOLVED, IGNORED

    public Finding() {
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getAnalysisId() {
        return analysisId;
    }

    public void setAnalysisId(String analysisId) {
        this.analysisId = analysisId;
    }

    public String getCategory() {
        return category;
    }

    public void setCategory(String category) {
        this.category = category;
    }

    public String getRuleId() {
        return ruleId;
    }

    public void setRuleId(String ruleId) {
        this.ruleId = ruleId;
    }

    public String getSeverity() {
        return severity;
    }

    public void setSeverity(String severity) {
        this.severity = severity;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getFilePath() {
        return filePath;
    }

    public void setFilePath(String filePath) {
        this.filePath = filePath;
    }

    public Integer getLineNumber() {
        return lineNumber;
    }

    public void setLineNumber(Integer lineNumber) {
        this.lineNumber = lineNumber;
    }

    public String getEvidence() {
        return evidence;
    }

    public void setEvidence(String evidence) {
        this.evidence = evidence;
    }

    public String getConfidence() {
        return confidence;
    }

    public void setConfidence(String confidence) {
        this.confidence = confidence;
    }

    public String getImpact() {
        return impact;
    }

    public void setImpact(String impact) {
        this.impact = impact;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }
}
